server/handler/auth.go

```diff
@@ -16,7 +16,7 @@ func EnsureAuthenticated(next http.HandlerFunc) http.HandlerFunc {
 			respondError(w, http.StatusInternalServerError, err.Error())
 			return
 		}
-		hasSession, err := sessionHelper.HasSession(r)
+		hasSession, err := sessionHelper.ValidSession(r)
```

Comment:
See the other comment regarding checking for SSO session when checking for a valid session.